Remove building and signing XPI file

Since we updated `web-ext` package, now we should specify addon id in manifest instead of provide it with flag for command. More details here: https://github.com/mozilla/web-ext/issues/3124

As result, we have problem when build a release:
```
Run mkdir ./AMO
  mkdir ./AMO
  npx web-ext sign --api-key "$AMO_KEY" --api-secret "$AMO_SECRET" --id "{e3fc2d33-09fc-4fe8-9331-d0a464698035}" --source-dir ./firefox --artifacts-dir ./AMO
  mv AMO/*.xpi AMO/linguist.xpi
  shell: /usr/bin/bash -e {0}
  env:
    AMO_KEY: ***
    AMO_SECRET: ***
  
web-ext sign
Sign the extension so it can be installed in Firefox
Options:
      --version             Show version number  [boolean]
  -s, --source-dir          Web extension source directory.  [string] [required] [default: "/home/runner/work/linguist/linguist/build"]
  -a, --artifacts-dir       Directory where artifacts will be saved.  [string] [required] [default: "/home/runner/work/linguist/linguist/build/web-ext-artifacts"]
  -v, --verbose             Show verbose output  [boolean]
  -i, --ignore-files        A list of glob patterns to define which files should be ignored. (Example: --ignore-files=path/to/first.js path/to/second.js "**/*.log")  [array]
      --no-input            Disable all features that require standard input  [boolean]
  -c, --config              Path to a CommonJS config file to set option defaults  [string]
      --config-discovery    Discover config files in home directory and working directory. Disable with --no-config-discovery.  [boolean] [default: true]
      --amo-base-url        Submission API URL prefix  [string] [required] [default: "https://addons.mozilla.org/api/v5/"]
      --api-key             API key (JWT issuer) from addons.mozilla.org  [string] [required]
      --api-secret          API secret (JWT secret) from addons.mozilla.org  [string] [required]
      --api-proxy           Use a proxy to access the signing API. Example: https://yourproxy:6000  [string]
      --timeout             Number of milliseconds to wait before giving up  [number]
      --approval-timeout    Number of milliseconds to wait for approval before giving up. Set to 0 to disable waiting for approval. Fallback to `timeout` if not set.  [number]
      --channel             The channel for which to sign the addon. Either 'listed' or 'unlisted'.  [string] [required]
      --amo-metadata        Path to a JSON file containing an object with metadata to be passed to the API. See https://addons-server.readthedocs.io/en/latest/topics/api/addons.html for details.  [string]
      --upload-source-code  Path to an archive file containing human readable source code of this submission, if the code in --source-dir has been processed to make it unreadable. See https://extensionworkshop.com/documentation/publish/source-code-submission/ for details.  [string]
  -h, --help                Show help  [boolean]
Unknown argument: id
Error: Process completed with exit code 1.
```

As temporary solution, we may remove building XPI file for serve a standalone addon.

Later we may add a specific target `firefox-standalone`, that will patch manifest file and add `"browser_specific_settings": {"gecko": {"id": "{e3fc2d33-09fc-4fe8-9331-d0a464698035}"}}`

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -35,7 +35,7 @@ jobs:
           AMO_SECRET: ${{ secrets.AMO_SECRET }}
         run: |
           mkdir ./AMO
-          npx web-ext sign --api-key "$AMO_KEY" --api-secret "$AMO_SECRET" --id "{e3fc2d33-09fc-4fe8-9331-d0a464698035}" --source-dir ./firefox-standalone --artifacts-dir ./AMO
+          npx web-ext sign --api-key "$AMO_KEY" --api-secret "$AMO_SECRET" --source-dir ./firefox-standalone --artifacts-dir ./AMO
           mv AMO/*.xpi AMO/linguist.xpi
 
       - name: Archive build files

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -28,14 +28,14 @@ jobs:
           npx crx pack chromium -o linguist.crx -p ./crx.pem
           ls -al ./
 
-      - name: Build XPI file for firefox
+      - name: Build XPI file for firefox-standalone
         working-directory: "./build"
         env:
           AMO_KEY: ${{ secrets.AMO_KEY }}
           AMO_SECRET: ${{ secrets.AMO_SECRET }}
         run: |
           mkdir ./AMO
-          npx web-ext sign --api-key "$AMO_KEY" --api-secret "$AMO_SECRET" --id "{e3fc2d33-09fc-4fe8-9331-d0a464698035}" --source-dir ./firefox --artifacts-dir ./AMO
+          npx web-ext sign --api-key "$AMO_KEY" --api-secret "$AMO_SECRET" --id "{e3fc2d33-09fc-4fe8-9331-d0a464698035}" --source-dir ./firefox-standalone --artifacts-dir ./AMO
           mv AMO/*.xpi AMO/linguist.xpi
 
       - name: Archive build files

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -38,7 +38,7 @@ jobs:
           key: ${{ steps.cache-thirdparty-restore.outputs.cache-primary-key }}
 
       - name: Build some targets for test
-        run: make prepare buildFirefox packAll lintBuilds
+        run: make prepare buildFirefox buildFirefoxStandalone packAll lintBuilds
 
       - name: Archive build files
         uses: actions/upload-artifact@v4

--- a/makefile
+++ b/makefile
@@ -20,10 +20,12 @@ buildThirdparty:
 buildAll:
 	mkdir -p ./build
 	chmod 777 ./build
-	${DOCKER_COMPOSE} run linguist make buildFirefox buildChromium buildChrome
+	${DOCKER_COMPOSE} run linguist make buildFirefox buildFirefoxStandalone buildChromium buildChrome
 
 buildFirefox:
 	NODE_ENV=production EXT_TARGET=firefox npx webpack-cli -c ./webpack.config.js
+buildFirefoxStandalone:
+	NODE_ENV=production EXT_TARGET=firefox-standalone npx webpack-cli -c ./webpack.config.js
 buildChromium:
 	NODE_ENV=production EXT_TARGET=chromium npx webpack-cli -c ./webpack.config.js
 buildChrome:

--- a/manifests/firefox-standalone.json
+++ b/manifests/firefox-standalone.json
@@ -1,0 +1,11 @@
+{
+	"browser_action": {
+		"theme_icons": [
+			{
+				"light": "static/logo/logo-icon-simple-light.png",
+				"dark": "static/logo/logo-icon-simple-dark.png",
+				"size": 32
+			}
+		]
+	}
+}

--- a/webpack.config.js
+++ b/webpack.config.js
@@ -110,14 +110,6 @@ module.exports = {
 
 						// Patch manifest with production overrides
 						const productionOverridesMap = {
-							firefox: {
-								// eslint-disable-next-line camelcase
-								browser_specific_settings: {
-									gecko: {
-										id: '{e5b6e4ac-ec96-44f5-b257-e4d3c8291b41}',
-									},
-								},
-							},
 							'firefox-standalone': {
 								// eslint-disable-next-line camelcase
 								browser_specific_settings: {

--- a/webpack.config.js
+++ b/webpack.config.js
@@ -108,6 +108,30 @@ module.exports = {
 							manifest = merge(manifest, targetManifest);
 						}
 
+						// Patch manifest with production overrides
+						const productionOverridesMap = {
+							firefox: {
+								// eslint-disable-next-line camelcase
+								browser_specific_settings: {
+									gecko: {
+										id: '{e5b6e4ac-ec96-44f5-b257-e4d3c8291b41}',
+									},
+								},
+							},
+							'firefox-standalone': {
+								// eslint-disable-next-line camelcase
+								browser_specific_settings: {
+									gecko: {
+										id: '{e3fc2d33-09fc-4fe8-9331-d0a464698035}',
+									},
+								},
+							},
+						};
+						if (isProduction && target in productionOverridesMap) {
+							const productionOverrides = productionOverridesMap[target];
+							manifest = merge(manifest, productionOverrides);
+						}
+
 						// Set version
 						manifest.version = package.version;
 

--- a/webpack.config.js
+++ b/webpack.config.js
@@ -18,7 +18,7 @@ const devtool = isProduction ? undefined : 'inline-source-map';
 const isFastBuild = !isProduction && process.env.FAST_BUILD === 'on';
 const isBundleAnalyzingEnabled = Boolean(process.env.DEBUG) || !isProduction;
 
-const targetsList = ['firefox', 'chromium', 'chrome'];
+const targetsList = ['firefox', 'firefox-standalone', 'chromium', 'chrome'];
 if (targetsList.indexOf(target) === -1) {
 	throw new Error(`Invalid target "${target}" in EXT_TARGET`);
 }

